### PR TITLE
doc: update release notes for 1.10

### PR DESCRIPTION
**Pull request by nashif**
Signed-off-by: Anas Nashif <anas.nashif@intel.com>

### CODE REVIEW

File: doc/release-notes-1.10.rst [hunk 1]
```diff
@@ -92,15 +102,23 @@ Build and Infrastructure
...
-Libraries
-*********
+Libraries / Subsystems
+***********************
+
+* Add random subsystem: Some "random" drivers are not drivers at all: they just
+  implement the function `sys_rand32_get()`.  Move those to a random subsystem
+  in preparation for a reorganization.
 
-* details ...
```
Comment (lpereira) [inline doc/release-notes-1.10.rst]: Please note here that TinyCrypt has been updated to 0.2.8.  No notable changes, since most of them were already integrated in Zephyr during the 1.9 release.

File: doc/release-notes-1.10.rst [hunk 2]
```diff
@@ -92,15 +102,23 @@ Build and Infrastructure
   You *will* need to port your own out-of-tree scripts and Makefiles to
   CMake.
 
-Libraries
-*********
+Libraries / Subsystems
+***********************
+
+* Add random subsystem: Some "random" drivers are not drivers at all: they just
```
Comment (lpereira) [inline doc/release-notes-1.10.rst]: I'd rephrase this as something like:

> The implementation for `sys_rand32_get()` function has been moved to a new "random" subsystem. There are new implementations for this function, one based in the Xoroshift128+ PRNG (using a hardware number generator to seed), and another that obtains random numbers directly from a hardware number generator driver. Hardware number generator drivers have been moved to a "drivers/entropy" directory; these drivers only expose the interface provided by `include/entropy.h`.

File: doc/release-notes-1.10.rst [hunk 3]
```diff
@@ -20,22 +20,39 @@ The following sections provide detailed lists of changes by component.
...
 ******
 
-* details ...
+* Power Management for nrf52 series SOC
 
 Drivers and Sensors
 *******************
 
 * timer: Add Support for TICKLESS KERNEL in xtensa_sys_timer
+* Rename `random` to `entropy`
```
Comment (lpereira) [inline doc/release-notes-1.10.rst]: I'd add a note that the I2C driver for ESP32 is known to be faulty.

File: doc/release-notes-1.10.rst [hunk 4]
```diff
@@ -20,22 +20,39 @@ The following sections provide detailed lists of changes by component.
 Kernel
 ******
 
-* details ...
+* Remove deprecated k_mem_pool_defrag code
+* add memory domain APIs Add the following application-facing memory domain
+  APIs:
+  * k_mem_domain_init() - to initialize a memory domain
```
Comment (dbkinder) [inline doc/release-notes-1.10.rst]: For brevity, you could just refer folks to the API documentation for these k_mem_domain functions:  http://docs.zephyrproject.org/api/kernel_api.html#memory-domain

File: doc/release-notes-1.10.rst [hunk 5]
```diff
@@ -20,22 +20,39 @@ The following sections provide detailed lists of changes by component.
...
+  * k_mem_domain_destroy() - to destroy a memory domain
+  * k_mem_domain_add_partition() - to add a partition into a domain
+  * k_mem_domain_remove_partition() - to remove a partition from a domain
+  * k_mem_domain_add_thread() - to add a thread into a domain
+  * k_mem_domain_remove_thread() - to remove a thread from a domain
+* add k_calloc() which uses kernel heap to implement traditional calloc()
+  semantics.
+* Introduce object validation mechanism: All system calls made from userspace
+  which involve pointers to kernel objects (including device drivers) will need
+  to have those pointers validated; userspace should never be able to crash the
```
Comment (dbkinder) [inline doc/release-notes-1.10.rst]: "should" would be better stated as "must" here (as in, "must never be able to..."

**Review by dbkinder — CHANGES_REQUESTED** [review]
suggested edits

File: doc/release-notes-1.10.rst (hunk 3 above)
Comment (nashif) [reply doc/release-notes-1.10.rst]: Please add this in a separate PR with more details.
